File: test/smoke.test.js
const fs = require('fs');
const path = require('path');
const { JSDOM } = require('jsdom');

const file = path.join(__dirname, '..', 'index.html');

JSDOM.fromFile(file, { runScripts: 'dangerously', resources: 'usable' }).then(dom => {
  const win = dom.window;
  // wait for load event or a short delay
  function runChecks(){
    try{
      const doc = win.document;
      const required = ['candidateSelect','addCandidate','goal','reality','options','will','save','sessionsList','exportCSV','printPdf'];
      const missing = required.filter(id => !doc.getElementById(id));
      if(missing.length){
        console.error('MISSING_ELEMENTS', missing);
        process.exit(2);
      }
      const funcs = ['loadSession','openEditModal','openHistoryModal'];
      const missingFuncs = funcs.filter(f => typeof win[f] !== 'function');
      if(missingFuncs.length){
        console.error('MISSING_FUNCTIONS', missingFuncs);
        process.exit(3);
      }
      console.log('OK: smoke tests passed');
      process.exit(0);
    }catch(err){
      console.error('ERROR', err);
      process.exit(1);
    }
  }

  if (win.document.readyState === 'complete') {
    runChecks();
  } else {
    win.addEventListener('load', runChecks);
    setTimeout(() => {
      // fallback
      if (win.document.readyState === 'complete') runChecks();
      else { console.error('TIMEOUT waiting for load'); process.exit(4); }
    }, 3000);
  }
}).catch(err=>{ console.error('JSDOM ERROR', err); process.exit(5); });

File: test/smoke_test.py
import re
import sys
from pathlib import Path

p = Path(__file__).resolve().parent.parent / 'index.html'
if not p.exists():
    print('index.html not found at', p)
    sys.exit(2)
text = p.read_text(encoding='utf-8')

required_ids = ['candidateSelect','addCandidate','goal','reality','options','will','save','sessionsList','exportCSV','printPdf']
missing = []
for idname in required_ids:
    if not re.search(r'id\s*=\s*["\']%s["\']' % re.escape(idname), text):
        missing.append(idname)

required_funcs = ['function loadSession','function openEditModal','function openHistoryModal']
missing_funcs = [f for f in required_funcs if f not in text]

if missing:
    print('MISSING_ELEMENTS', missing)
if missing_funcs:
    print('MISSING_FUNCTIONS', missing_funcs)

if not missing and not missing_funcs:
    print('OK: static smoke tests passed')
    sys.exit(0)
else:
    sys.exit(1)
